NICE.Identity.Management/ClientApp/src/views/EditOrganisation/EditOrganisation.tsx

```diff
@@ -0,0 +1,309 @@
+import React, { Component } from "react";
+import { RouteComponentProps, Link, Redirect } from "react-router-dom";
+import { Breadcrumbs, Breadcrumb } from "@nice-digital/nds-breadcrumbs";
+import { Button } from "@nice-digital/nds-button";
+import { Grid, GridItem } from "@nice-digital/nds-grid";
+import { PageHeader } from "@nice-digital/nds-page-header";
+import { Input } from "@nice-digital/nds-input";
+import { Endpoints } from "../../data/endpoints";
+import { fetchData } from "../../helpers/fetchData";
+import { isDataError } from "../../helpers/isDataError";
+import { ErrorMessage } from "../../components/ErrorMessage/ErrorMessage";
+import { OrganisationType } from "../../models/types";
+
+type TParams = { id: string };
+
+type EditOrganisationProps = Record<string, unknown> &
+	RouteComponentProps<TParams>;
+
+type EditOrganisationState = {
+	validationError: boolean;
+	validationErrorMessage: string;
+	fetchedOrgNameFound: boolean;
+	organisation: OrganisationType;
+	formName: string;
+	isLoading: boolean;
+	hasSubmitted: boolean;
+	error?: Error;
+	isSaveButtonLoading: boolean;
+};
+export class EditOrganisation extends Component<
+	EditOrganisationProps,
+	EditOrganisationState
+> {
+	constructor(props: EditOrganisationProps) {
+		super(props);
+
+		this.state = {
+			validationError: false,
+			validationErrorMessage:
+				"Organisation name should be alphanumeric and be between 2-100 characters",
+			fetchedOrgNameFound: false,
+			organisation: {} as OrganisationType,
+			formName: "",
+			isLoading: false,
+			hasSubmitted: false,
+			isSaveButtonLoading: false,
+		};
+
+		document.title = "NICE Accounts - Edit organisation";
+	}
+
+	typingTimer = 0;
+
+	async componentDidMount(): Promise<void> {
+		this.setState({ isLoading: true });
+
+		const organisation = await fetchData(
+			Endpoints.organisation(this.props.match.params.id),
+		);
+
+		if (isDataError(organisation)) {
+			this.setState({ error: organisation });
+		} else {
+			this.setState({ formName: organisation.name });
+		}
+
+		this.setState({ organisation, isLoading: false });
+	}
+
+	checkOrgName = async (formName: string): Promise<void> => {
+		const orgName = this.state.organisation.name.toLowerCase();
+		let fetchedOrgNameFound = false;
+		formName = formName.toLowerCase();
+
+		if (formName !== orgName) {
+			const fetchOrgName = await fetchData(
+				`${Endpoints.organisationsList}?q=${formName}`,
+			);
+
+			// should there be error handling??
```

Comment:
there should be error handling, can this be wrapped in a try catch?

Reply:
😆 Bloody pesky comment.